test/test_transforms.py

```diff
@@ -673,3 +674,120 @@ def test_bert_tokenizer_save_load(self, test_scripting, do_lower_case, return_to
             torch.save(tokenizer, tokenizer_path)
             loaded_tokenizer = torch.load(tokenizer_path)
             self._bert_tokenizer((loaded_tokenizer), do_lower_case=do_lower_case)
+
+
+class TestRegexTokenizer(TorchtextTestCase):
+    def test_regex_tokenizer(self):
+        test_sample = "'\".<br />,()!?;:   Basic Regex Tokenization for a Line of Text   '\".<br />,()!?;:"
+        ref_results = [
+            "'",
+            ".",
+            ",",
+            "(",
+            ")",
+            "!",
+            "?",
+            "Basic",
+            "Regex",
+            "Tokenization",
+            "for",
+            "a",
+            "Line",
+            "of",
+            "Text",
+            "'",
+            ".",
+            ",",
+            "(",
+            ")",
+            "!",
+            "?",
+        ]
+        patterns_list = [
+            (r"\'", " '  "),
+            (r"\"", ""),
+            (r"\.", " . "),
+            (r"<br \/>", " "),
+            (r",", " , "),
+            (r"\(", " ( "),
+            (r"\)", " ) "),
+            (r"\!", " ! "),
+            (r"\?", " ? "),
+            (r"\;", " "),
+            (r"\:", " "),
+            (r"\s+", " "),
+        ]
+
+        r_tokenizer = RegexTokenizer(patterns_list)
+        eager_tokens = r_tokenizer(test_sample)
+
+        jit_r_tokenizer = torch.jit.script(r_tokenizer)
+        jit_tokens = jit_r_tokenizer(test_sample)
+
+        assert not r_tokenizer.is_jitable
+        # Call the __prepare_scriptable__() func and convert the operator to the torchbind version
+        # We don't expect users to use the torchbind version on eager mode but still need a CI test here.
+        assert r_tokenizer.__prepare_scriptable__().is_jitable
+
+        self.assertEqual(eager_tokens, ref_results)
+        self.assertEqual(jit_tokens, ref_results)
+
+    def test_regex_tokenizer_save_load(self):
+        test_sample = "'\".<br />,()!?;:   Basic Regex Tokenization for a Line of Text   '\".<br />,()!?;:"
+        ref_results = [
+            "'",
+            ".",
+            ",",
+            "(",
+            ")",
+            "!",
+            "?",
+            "Basic",
+            "Regex",
+            "Tokenization",
+            "for",
+            "a",
+            "Line",
+            "of",
+            "Text",
+            "'",
+            ".",
+            ",",
+            "(",
+            ")",
+            "!",
+            "?",
+        ]
+        patterns_list = [
+            (r"\'", " '  "),
+            (r"\"", ""),
+            (r"\.", " . "),
+            (r"<br \/>", " "),
+            (r",", " , "),
+            (r"\(", " ( "),
+            (r"\)", " ) "),
+            (r"\!", " ! "),
+            (r"\?", " ? "),
+            (r"\;", " "),
+            (r"\:", " "),
+            (r"\s+", " "),
+        ]
```

Comment:
Since we reuse `test_sample`, `ref_results`, and `patterns_list` across multiple tests, can we try to move them out of the test functions and make them class variables. This way we can avoid code duplication